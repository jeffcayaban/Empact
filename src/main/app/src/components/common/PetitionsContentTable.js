import React from "react";
import {Table} from "react-bootstrap";
import TableFooter from "./TableFooter";
import NoPetitionsAndExploreAlert from "./alerts/NoPetitionsAndExploreAlert";
import PetitionEntry from "./PetitionEntry";

/**
 * Displays the table of petitions.
 */

class PetitionsContentTable extends React.Component {

    render() {
        const { petitions, pageNumber, totalPages, showFollowUp, loadFromServer } = this.props;

        if (Array.isArray(petitions) && petitions.length) {
            // If there are petitions
            return (
                <div>
                    <Table bordered hover>
                        <tbody>
                        { petitions.map((petition, index) =>
                            <PetitionEntry
                                showCreatorName={true}
                                key={index}
                                petition={petition}
                                showCreationDetails={true}
                            />)
                        }
                        </tbody>
                    </Table>
                    <div style={{ paddingTop: '10px' }}>
                        <TableFooter
                            pageNumber={pageNumber}
                            totalPages={totalPages}
                            onPrevious={() => loadFromServer(pageNumber - 1)}
                            onNext={() => loadFromServer(pageNumber + 1)}
                        />
                    </div>
                </div>
            );

        } else {
            // If there are NO petitions.
            return <NoPetitionsAndExploreAlert showFollowUp={showFollowUp} />;
        }
    }

}

export default PetitionsContentTable;