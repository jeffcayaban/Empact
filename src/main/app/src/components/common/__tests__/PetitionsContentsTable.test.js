import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import PetitionsContentTable from "../PetitionsContentTable";
import MockPetitions from '../../../other/mock-data/MockGetPetitionsResponse';

describe('PetitionsContentTable', () => {

    it('should render successfully given there exists petitions', () => {
        const component = shallow(<PetitionsContentTable
            petitions={MockPetitions.content}
            pageNumber={MockPetitions.page}
            totalPages={MockPetitions.totalPages}
            showFollowUp={true}
            loadFromServer={jest.fn()}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render an alert given there exists no petitions', () => {
        const component = shallow(<PetitionsContentTable
            petitions={[]}
            pageNumber={MockPetitions.page}
            totalPages={MockPetitions.totalPages}
            showFollowUp={true}
            loadFromServer={jest.fn()}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

});
