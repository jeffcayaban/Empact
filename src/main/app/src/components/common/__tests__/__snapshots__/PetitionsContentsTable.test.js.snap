// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionsContentTable should render an alert given there exists no petitions 1`] = `
<NoPetitionsAndExploreAlert
  showFollowUp={true}
/>
`;

exports[`PetitionsContentTable should render successfully given there exists petitions 1`] = `
<div>
  <Table
    bordered={true}
    bsClass="table"
    condensed={false}
    hover={true}
    responsive={false}
    striped={false}
  >
    <tbody>
      <withRouter(PetitionEntry)
        key="0"
        petition={
          Object {
            "action": "TestAction",
            "closingDateTime": "2019-02-15T13:04:27Z",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:29Z",
            "goal": "TestGoal",
            "id": 1,
            "isAnonymous": false,
            "isWinning": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
            "situation": "TestSituation",
            "title": "TestTitle",
            "value": "TestValue",
          }
        }
        showCreationDetails={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={1}
    />
  </div>
</div>
`;
